e2e/disruptors/pod_e2e_test.go

```diff
@@ -154,16 +157,45 @@ func Test_PodDisruptor(t *testing.T) {
 				go func() {
 					err := tc.injector(disruptor)
 					if err != nil {
-						t.Logf("failed to setup disruptor: %v", err)
+						t.Errorf("failed to setup disruptor: %v", err)
 						return
 					}
 				}()
 
-				err = tc.check.Verify(k8s, cluster.Ingress(), namespace)
-				if err != nil {
-					t.Errorf("failed to access service: %v", err)
-					return
-				}
+				t.Run("on ingress", func(t *testing.T) {
+					t.Parallel()
+
+					err := tc.check.Verify(k8s, cluster.Ingress(), namespace)
+					if err != nil {
+						t.Fatalf("failed to access service: %v", err)
+						return
+					}
+				})
+
+				t.Run("on port-forward", func(t *testing.T) {
```

Comment:
I agree, I didn't like it a lot either. The alternative though was to duplicate a lot of the setup code. If you have other ideas or preferences I'm all ears :)

Reply:
I'm not sure there's "a lot" to be duplicated as most of the setup code has been refactored into the helper functions.
Besides, I will say it is better to repeat ourselves in tests than force together tests not to duplicate boilerplate code (that we can refactor later).  I always found over-generalizing tests a dangerous path because It forces the reader to understand the purpose of each test, instead of each test being explicit about its own purpose.

Reply:
> I always found over-generalizing tests a dangerous path

I definitely agree, at some point if your tables start containing too many fields and function pointers it very quickly becomes unreadable.

In this case thought, I wouldn't say we're generalizing: We want to test exactly the same cases in both scenarios, as (for now) all of them should work, so I'm not fully sold on us being over-generalizing here. Would you propose to copy-paste the whole test instead? I cannot spot any line that should be different 🤔

Reply:
> We want to test exactly the same cases in both scenarios, 
I don't think we should. That's the point. There's so much more to testing each fault injection than testing the traffic redirection. This is why we need an e2e test at least for each fault, at least.

However, the changes introduced in this PR are mostly independent of all these.

I Would only test one fault injection to verify the traffic redirection works as expected in the given scenario.  There's no value in my opinion in testing more than this in an e2e test.

So, what I'm suggesting is to create a separate test only with a test case instead of duplicating every e2e test on both scenarios. 

I'm even open to moving these tests to the `e2e/agent` is that makes more sense.

Reply:
> at some point if your tables start containing too many fields and function pointers it very quickly becomes unreadable
Hopefully, we haven't arrived to this point. In any case, I've been considering using generics to avoid this, but I'm unsure how

Reply:
Okay, I think I get what you mean now:
1. Keep the `e2e/disruptor` tests as they are, with the intent of expanding them to different faults (e.g. delays) later
2. Add a more specific test only for the redirection, and add only to this one both the ingress/pfw scenarios.

If I got that correctly, that makes sense to me. I'll start working on that.